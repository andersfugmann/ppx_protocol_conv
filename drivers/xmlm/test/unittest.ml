module Driver = struct
  let name = "xmlm"
  let serialize t = Ezxmlm.to_string [t]
  let deserialize t = Ezxmlm.from_string t |> snd |> List.hd
  include Protocol_conv_xmlm.Xmlm
  let of_driver_exn = of_xmlm_exn
  let of_driver = of_xmlm
  let to_driver = to_xmlm
  let sexp_of_t : t -> Sexplib.Sexp.t = fun t ->
    Sexplib.Std.sexp_of_string (to_string_hum t)
end
module Unittest = Test.Unittest.Make(Driver)
let () = Unittest.run()
