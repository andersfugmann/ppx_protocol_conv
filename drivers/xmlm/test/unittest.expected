=== xmlm.SingleElem ===
<l/>
=== xmlm.SingleElem ===
<l><p>2</p></l>
=== xmlm.Longarray ===
<l><p>4</p><p>2</p><p>3</p><p>1</p></l>
=== xmlm.EmptyInsideRec ===
<record><c>c</c><a>a</a></record>
=== xmlm.SingleInsideRec ===
<record><c>c</c><V>2</V><a>a</a></record>
=== xmlm.MultiInsideRec ===
<record><c>c</c><V>4</V><V>2</V><V>3</V><V>1</V><a>a</a></record>
=== xmlm.ArrayOfArrays ===
<record><a><p>2</p><p>3</p></a><a><p>4</p><p>5</p></a></record>
=== xmlm.ArrayOfArrays2 ===
<l><l/><l><l/><l><p>2</p></l><l><p>3</p><p>4</p></l></l><l><l/></l><l><l><p>2</p></l></l></l>
=== xmlm.Tuple ===
<l><record><t3><t1>200</t1><t0>100</t0></t3><t3><t1>400</t1><t0>300</t0></t3><t3><t1>600</t1><t0>500</t0></t3><t2>s50</t2><t2>s60</t2><t2>s70</t2><t1>20</t1><t1>30</t1><t1>40</t1><t0>10</t0></record><record><t3><t1>201</t1><t0>101</t0></t3><t3><t1>401</t1><t0>301</t0></t3><t3><t1>601</t1><t0>501</t0></t3><t2>s51</t2><t2>s61</t2><t2>s71</t2><t1>21</t1><t1>31</t1><t1>41</t1><t0>11</t0></record><record><t3><t1>202</t1><t0>102</t0></t3><t3><t1>402</t1><t0>302</t0></t3><t3><t1>602</t1><t0>502</t0></t3><t2>s52</t2><t2>s62</t2><t2>s72</t2><t1>22</t1><t1>32</t1><t1>42</t1><t0>12</t0></record><record><t3><t1>203</t1><t0>103</t0></t3><t3><t1>403</t1><t0>303</t0></t3><t3><t1>603</t1><t0>503</t0></t3><t2>s53</t2><t2>s63</t2><t2>s73</t2><t1>23</t1><t1>33</t1><t1>43</t1><t0>13</t0></record></l>
=== xmlm.Any ===
<record><u __name="record"><ua>7</ua></u><v __name="variant">B<l><p>5</p><p>6</p><p>7</p></l><l><p>10</p><p>11</p><p>12</p></l></v><z>101</z><record><y>string</y><x>5</x></record><varray>B<l><p>5</p><p>6</p><p>7</p></l><l><p>10</p><p>11</p><p>12</p></l></varray><varray>B<l><p>5</p><p>6</p><p>7</p></l><l><p>10</p><p>11</p><p>12</p></l></varray><varray>B<l><p>5</p><p>6</p><p>7</p></l><l><p>10</p><p>11</p><p>12</p></l></varray><vlist>B<l><p>5</p><p>6</p><p>7</p></l><l><p>10</p><p>11</p><p>12</p></l></vlist><vlist>B<l><p>5</p><p>6</p><p>7</p></l><l><p>10</p><p>11</p><p>12</p></l></vlist><vlist>B<l><p>5</p><p>6</p><p>7</p></l><l><p>10</p><p>11</p><p>12</p></l></vlist><tuple><t2>false</t2><t1>protocol</t1><t0>5</t0></tuple><intref>4</intref><intoption>100</intoption><intlist>3</intlist><intlist>4</intlist><intlist>5</intlist><bytes>bytes</bytes><string2>1</string2><string>string</string><float>3.14</float><nativeint>20</nativeint><int64>10</int64><int32>5</int32><int>2</int><char>x</char><bool>true</bool></record>
=== xmlm.Record ===
<record><t_il>1000</t_il><t_il>1001</t_il><t_il>1002</t_il><t_tl><t2>s100</t2><t2>s101</t2><t1>101</t1><t0>100</t0></t_tl><t_tl><t2>s110</t2><t2>s111</t2><t1>111</t1><t0>110</t0></t_tl><t_tl><t2>s120</t2><t2>s121</t2><t1>121</t1><t0>120</t0></t_tl><t_tl><t2>s130</t2><t2>s131</t2><t1>131</t1><t0>130</t0></t_tl><t_tl><t2>s140</t2><t2>s141</t2><t1>141</t1><t0>140</t0></t_tl><t_t><t2>s100</t2><t2>s101</t2><t1>101</t1><t0>100</t0></t_t><t_i>1000</t_i><t_b><b_al><a_string>s7</a_string><a_int>7</a_int></b_al><b_al><a_string>s8</a_string><a_int>8</a_int></b_al><b_al><a_string>s9</a_string><a_int>9</a_int></b_al><b_a><a_string>s6</a_string><a_int>6</a_int></b_a><b_string>s5</b_string><b_int>5</b_int></t_b><t_al><a_string>s2</a_string><a_int>2</a_int></t_al><t_al><a_string>s3</a_string><a_int>3</a_int></t_al><t_al><a_string>s4</a_string><a_int>4</a_int></t_al><t_a><a_string>s1</a_string><a_int>1</a_int></t_a></record>
=== xmlm.list ===
<record><a>1</a><a>2</a><a>3</a></record>
=== xmlm.Lists ===
<record><l>A<l><p>1</p><p>2</p><p>3</p></l></l><l>B<l><l><p>1</p><p>2</p></l><l><p>3</p><p>4</p><p>5</p></l><l><p>2</p></l></l><l><p>3</p><p>1</p></l><p>5</p></l><l>C<l><p>1</p><p>2</p><p>3</p></l><l><p>3</p><p>4</p><p>5</p></l></l><c>100</c><c>101</c><c>102</c><c>103</c><b><t1>10</t1><t1>20</t1><t1>30</t1><t1>40</t1><t0>8</t0><t0>9</t0></b><a><p>1</p><p>2</p><p>3</p></a><a/><a><p>10</p><p>20</p><p>30</p><p>40</p></a><a><p>100</p><p>101</p></a></record>
=== xmlm.array ===
<record><a>1</a><a>2</a><a>3</a></record>
=== xmlm.EmptyList ===
<l/>
=== xmlm.SingleElem ===
<l><p>2</p></l>
=== xmlm.Longlist ===
<l><p>4</p><p>3</p><p>2</p><p>1</p></l>
=== xmlm.EmptyInsideRec ===
<record><c>c</c><a>a</a></record>
=== xmlm.SingleInsideRec ===
<record><c>c</c><V>2</V><a>a</a></record>
=== xmlm.MultiInsideRec ===
<record><c>c</c><V>4</V><V>2</V><V>3</V><V>1</V><a>a</a></record>
=== xmlm.ListOfLists ===
<record><a><p>2</p><p>3</p></a><a><p>4</p><p>5</p></a></record>
=== xmlm.ListOfLists2 ===
<l><l/><l><l/><l><p>2</p></l><l><p>3</p><p>4</p></l></l><l><l/></l><l><l><p>2</p></l></l></l>
=== xmlm.Nonrec ===
<variant>A<variant>Cons<p>4</p><variant>Cons<p>3</p><variant>Nil</variant></variant></variant></variant>
=== xmlm.Nonrec2 ===
<variant>Cons<p>4</p><variant>Cons<p>3</p><variant>Nil</variant></variant></variant>
=== xmlm.None ===
<__option/>
=== xmlm.Some None ===
<__option><__option/></__option>
=== xmlm.Some Some None ===
<__option><__option><__option/></__option></__option>
=== xmlm.Some Some Some Unit ===
<p>()</p>
=== xmlm.simple ===
<p>5</p>
=== xmlm.record ===
<record><a>5</a></record>
=== xmlm.multiple ===
<record><t2>true</t2><t1>5</t1><t0>5</t0></record>
=== xmlm.reference ===
<record><a><a>5</a></a></record>
=== xmlm.recursive ===
<record><c><b><a>5</a></b></c></record>
=== xmlm.Simple ===
<l><variant>A</variant><variant>B<p>5</p></variant><variant>C<record><t1>7</t1><t0>6</t0></record></variant><variant>D<record><t1>9</t1><t0>8</t0></record></variant></l>
=== xmlm.Tree ===
<variant>Node<record><t2>Leaf</t2><t1>10</t1><t0>Node<record><t2>Leaf</t2><t1>3</t1><t0>Leaf</t0></record></t0></record></variant>
=== xmlm.MutualRecursion ===
<variant>T1<variant>V<variant>T<variant>V<variant>V1<variant>V1<variant>V1<variant>V0<p>5</p></variant></variant></variant></variant></variant></variant></variant></variant>
=== xmlm.InsideRec ===
<record><c>c</c><V>A</V><a>a</a></record>
=== xmlm.RecordList ===
<record><objects><key>1</key></objects><objects><key>2</key></objects></record>
=== xmlm.SimpleRecord ===
<record><HostId>SDsd</HostId><RequestId>sdfsd</RequestId><Endpoint/><Bucket/><Message>Message</Message><Code>Error</Code></record>
=== xmlm.Test_sig ===
<record><x>A<record><t4>7</t4><t3>B<p>0.7</p></t3><t2>7</t2><t1>7</t1><t0>A<p>7</p></t0></record></x></record>
=== xmlm.Test_sig2 ===
<record><t6>A<p>1</p></t6><t5><c>3.0</c><b>2.</b><a>1</a></t5><t4>A<p>1</p></t4><t3>()</t3><t2>3.0</t2><t1>2.</t1><t0>1</t0></record>
=== xmlm.S3 ===
<record><Contents><ETag>Etag</ETag><StorageClass>STANDARD</StorageClass></Contents><Prefix>prefix</Prefix></record>
=== xmlm.Types ===
<record><baz><y_yd>Variant_two1<p>1</p></y_yd><y_yc><t1>100</t1><t1>200</t1><t1>300</t1><t0>three</t0></y_yc><y_b><t1>10</t1><t1>20</t1><t1>30</t1><t0>two</t0></y_b><y_a>2</y_a></baz><bar>true</bar><foo>1</foo></record>
=== xmlm.Some Some Some true ===
<p>true</p>
=== xmlm.Some Some None ===
<__option><__option><__option/></__option></__option>
=== xmlm.Some None ===
<__option><__option/></__option>
=== xmlm.None ===
<__option/>
=== xmlm.Some Some Some true ===
<record><a>true</a></record>
=== xmlm.Some Some None ===
<record><a><__option><__option/></__option></a></record>
=== xmlm.Some None ===
<record><a><__option/></a></record>
=== xmlm.None ===
<record><a/></record>
=== xmlm.unit option option list option option ===
<l><p>()</p><__option><__option/></__option><__option/></l>
=== xmlm.confuse deserialization by using reserved word ===
<record><o><option>true</option></o></record>
=== xmlm.Simple ===
<l><variant>A</variant><variant>B<p>5</p></variant><variant>C<p>6</p><p>7</p></variant><variant>D<record><t1>9</t1><t0>8</t0></record></variant></l>
=== xmlm.Tuple ===
<variant>A<record><t1>4</t1><t0>3</t0></record></variant>
=== xmlm.Tree ===
<variant>Node<variant>Node<variant>Leaf</variant><p>3</p><variant>Leaf</variant></variant><p>10</p><variant>Leaf</variant></variant>
=== xmlm.MutualRecursion ===
<variant>T1<variant>V<variant>T<variant>V<variant>V1<variant>V1<variant>V1<variant>V0<p>5</p></variant></variant></variant></variant></variant></variant></variant></variant>
=== xmlm.InsideRec ===
<record><c>c</c><V>A</V><a>a</a></record>
=== xmlm.InlineRecord ===
<variant>A<record><a>a</a></record></variant>
=== xmlm.InlineRecord2 ===
<variant>aa<record><b>aa<record><b>B<p>5</p></b><A>a</A></record></b><A>a</A></record></variant>
=== xmlm.Poly ===
<variant>aaa<p>5</p></variant>
=== xmlm.Option.Ok ===
<variant>Ok<p>2</p></variant>
=== xmlm.Option.Error ===
<variant>Error<p>Error string</p></variant>
